pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial.md

```diff
@@ -1,231 +1,291 @@
+  <form method="post" action-xhr="#" target="_top" class="card">
+    ...
+  </form>
 </amp-script>
 ```
 
-The `src` attribute points to the filepath `http://localhost:8080/js/script.js`. Create a directory titled `js` in the `public` directory and add the `script.js` file.
+Our `<amp-script>` includes the attribute `sandbox="allow-forms"`. That tells AMP it's ok for the script to modify the content of the form.
 
+Since AMP aims to guarantee a fast, visually stable user experience, it won't let our JavaScript make unrestricted changes to the DOM at any time. Your JavaScript can make more changes if the size of the `<amp-script>` component can't change. It also allows more substantial changes after a user interaction. You can find details in [the reference documentation](../../../documentation/components/reference/amp-script.md). For this tutorial, it suffices to know that we've specified a `layout` type that isn't `container`, and we've used HTML attributes to lock down the component's size. This means that any DOM manipulations are restricted to a certain area of the page.
 
-# Inject custom script
+If you're using the [AMP validator Chrome extension](https://chrome.google.com/webstore/detail/amp-validator/nmoffdblmcmgeicmolmhobpoocbbmknc) you will now see an error message:
 
-Open the newly created `script.js` file and add `console.log("amp-script here")`. Reload the page and open the [DevTools console](https://developers.google.com/web/tools/chrome-devtools/) to verify it successfully logged "amp-script here".
+{{ image('/static/img/docs/tutorials/custom-javascript-tutorial/relative-url-error.png', 600, 177, layout='intrinsic', alt='Error about relative URL', align='center' ) }}
 
-Imported script logic from the `amp-script` `src` attribute is debugged in the console, same as JavaScript inside non-AMP pages.
+[tip type="note"]
+If you don't have this extension, append `#development=1` to your URL, and AMP will output validation errors to your Console.
+[/tip]
 
-# Add logic
+What does this mean? AMP components that specify a file require an absolute URL. This is necessary because our AMP document might be served from [an AMP cache](https://blog.amp.dev/2017/01/13/why-amp-caches-exist). When that happens, unless our site uses a [signed exchange](https://developers.google.com/web/updates/2018/11/signed-exchanges), our content will be served from a different domain. (To learn more about AMP caches and signed exchanges, [see here](../optimize-measure/signed-exchange.md).) Using an absolute URL like `http://localhost/js/validate.js` would fix this. But AMP also requires the use of [HTTPS](https://developers.google.com/web/fundamentals/security/encrypt-in-transit/why-https). So we would still get a validation error, and setting up SSL on our local webserver is outside the scope of this tutorial. If you want to do it, you can follow the instructions in [this post](https://timonweb.com/posts/running-expressjs-server-over-https/).
```

Comment:
Please remove the general part about absolute URLs as it's not correct. amp-script requires absolute URLs (which I'm not sure why it's needed), but not all components.

Reply:
Will look into the reasoning. Meanwhile, I've removed this.